waku/v2/peermanager/topic_event_handler.go

```diff
@@ -118,6 +120,37 @@ func (pm *PeerManager) handlerPeerTopicEvent(peerEvt relay.EvtPeerTopic) {
 	wps := pm.host.Peerstore().(*wps.WakuPeerstoreImpl)
 	peerID := peerEvt.PeerID
 	if peerEvt.State == relay.PEER_JOINED {
+		// Metadata verification is done only if a clusterID is specified
```

Comment:
Not sure how this is used but perhaps its more efficient in the `ConnectionGater`?

Example https://github.com/prysmaticlabs/prysm/blob/develop/beacon-chain/p2p/connection_gater.go#L43C19-L61

Reply:
I'll try. Not sure if i can dial a node within the connection gater.

Reply:
mm interesting. actually that could be and endless recursive loop? since that dial will be handled by the gater of the other node, that will dial back and will be handled by the gater, that again will dial and will be handled by the gater?

unsure if its the case.

Reply:
I confirm that it gets stuck into a recursive loop.  :-1: